tests/testutil/service.go

```diff
@@ -11,21 +11,24 @@ import (
 
 	"k8s.io/apimachinery/pkg/util/wait"
 
+	"github.com/knative/pkg/apis"
 	riov1 "github.com/rancher/rio/pkg/apis/rio.cattle.io/v1"
+	tektonv1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
 )
 
 type TestService struct {
 	Name    string // namespace/name:version
 	AppName string // namespace/name
 	App     riov1.App
 	Service riov1.Service
+	Build   tektonv1alpha1.TaskRun
 	Version string
 	T       *testing.T
 }
 
 // Create generates a new rio service, named randomly in the testing namespace, and
 // returns a new TestService with it attached. Guarantees ready state but not live endpoint
-func (ts *TestService) Create(t *testing.T, source string) {
+func (ts *TestService) Create(t *testing.T, source string, args ...string) {
```

Comment:
This is being done in https://github.com/rancher/rio/pull/548 also, make sure they match up.

Reply:
That PR did this in a much more elegant way. I'll update this to do the same. Need to figure out a better way to default port to 80/http. It seems to work on our docker images but fails on GH builds. Potential solution is including the argument if the source isn't from GH. I'll do that and we can change if we see issues.